### DISCUSSION TITLE
BFD Export Option/BB2RIF Exporter Code Maps

### DISCUSSION THREAD

**Discussion by rileeki** [synthetichealth/synthea#1389] [Q&A] [answered]
I'm interested in using the BFD export option to get output that looks like [Medicare's Synthetic RIF files](https://data.cms.gov/sites/default/files/2023-05/d51e1218-68c3-4c7c-9598-0b81f22fe903/User%20Guide%20-%20CMS%20Synthetic%20RIF%20Files%20May%202023_AM508_v2.pdf).  

I was hoping it'd be as easy as just setting line 60 of the synthea.properties file to True, but it looks like I need to provide a bunch of code maps in .json format (eg "export/drg_code_map.json" and "export/hcpcs_code_map.json").  

Am I right in guessing that you do not provide these because of licensing issues?  If we can get our own licenses, would we be able to request these files or create them ourselves?  What is the general format and content of these code map files?

**Comment by hadleynet** [comment, marked as answer]
Yes re licensing issues being the reason we don't publish the mapping file.

The mapping files are structured like this

```
{
  "70536003": [ // synthea SNOMED code
    {
      "code": "XXXXX", // first mapped CPT code
      "description": "description of code",
      "weight": "10" // all weights are summed and then individual weights are used to drive the distribution of codes so this one is twice as likely to be used as the next one
    },
    {
      "code": "YYYYY", // second mapped CPT code
      "description": "description of code",
      "weight": "5"
    },
    ...
  ],
  ...
}
```

If you create a set of empty mapping files (just "{}") and then run the exporter, it will generate a missing_codes.csv file that will give you a list of all of the codes that need to be mapped in each of the files.

**Reply by rileeki** [reply]
Thank you so much!  That helps a ton.
